core/src/main/java/org/ldaptive/auth/SearchDnResolver.java

```diff
@@ -262,7 +287,21 @@ public String resolve(final User user)
    */
   protected String resolveDn(final LdapEntry entry)
   {
-    return entry.getDn();
+    String dn = entry.getDn();
```

Comment:
Configuration for `resolveFromAttribute` should be either/or. That is, if it's turned on it should either produce a DN from an attribute or return null. Not fallback to the entry DN.

Reply:
I have rewriten this part and added a new method to make the rule much more explicit.